### ISSUE TITLE
Fedora 38 - fcitx5 unresponsive on fresh boot

### ISSUE THREAD

**Issue by DonKatsu** [fcitx/fcitx5#803]
**Describe the bug**
After booting, trying to switch input methods via hotkey does not work. Forcing an input method (KKC for example) through the notification icon and trying to type with it still uses the default input method.
After right clicking the notification icon and clicking Restart, fcitx5 works as expected.
Logging out and in again after restarting fcitx5, it continues to work as expected.

**To Reproduce**
1. Restart system
2. Try to switch input methods

**Expected behavior**
Normal functionality.

**Desktop (please complete the following information):**
 - Desktop: KDE 5.27.5
 - Display server type: Wayland
 - [fcitx5-diagnose.txt](https://github.com/fcitx/fcitx5/files/11600665/fcitx5-diagnose.txt)

**Additional context**
After reaching the desktop from a restart, performing this grep returns the following:
```
[~]$ journalctl -b | grep fcitx5
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.304801 instance.cpp:1275] Override Enabled Addons: {}
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.304839 instance.cpp:1276] Override Disabled Addons: {}
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.307130 addonmanager.cpp:191] Loaded addon xcb
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.307895 addonmanager.cpp:191] Loaded addon quickphrase
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.308013 addonmanager.cpp:191] Loaded addon imselector
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311079 addonmanager.cpp:191] Loaded addon keyboard
May 30 08:32:44 system fcitx5[3103]: E2023-05-30 08:32:44.311357 addonloader.cpp:61] Failed to create addon: dbus Unable to request dbus name. Is there another fcitx already running?
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311372 addonmanager.cpp:188] Could not load addon dbus
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311403 addonmanager.cpp:297] Unloading addon keyboard
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311464 addonmanager.cpp:297] Unloading addon imselector
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311480 addonmanager.cpp:297] Unloading addon quickphrase
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311598 addonmanager.cpp:297] Unloading addon xcb
May 30 08:32:44 system fcitx5[3103]: I2023-05-30 08:32:44.311609 inputcontextmanager.cpp:358] All display connections are gone, exit now.
```
Additional restarts of fcitx5 do not return any new results.

**Comment by wengxt** [comment]
fcitx5-diagnose indicates that you are quite likely didn't set environment variable to your whole session correctly and some are set to ibus.

How did you set environment variables (GTK_IM_MODULE, QT_IM_MODULE, XMODIFIERS)

do you have any configuration under ~/.config/environment.d/ ?

**Comment by DonKatsu** [comment]
Actually, you can probably disregard a lot of that last post...

Skimming the wiki and such again, I realized what I probably did wrong.
When I was going through some system settings I saw fcitx in the virtual keyboards section and set it to that. I was using fcitx5-autostart up until then.

So I had both the fcitx5 virtual keyboard and the autostart package, causing a conflict at startup I guess.
Turning off the virtual keyboard and using only the autostart package, or uninstalling the autostart package and using the virtual keyboard works as expected.

If such a conflict is already known, sorry about that.

**Comment by wengxt** [comment]
xdg autostart and kwin's virtualkeyboard may theoretically conflict, if you want to use zwp_input_method protocol for input method it has to be the one started by kwin. Though I personally never had an issue in my system, maybe my harddisk loads kwin's input method faster.

I had a plan in my mind to make kwin's fcitx start to use fcitx5-wayland-launcher, to make it use dbus activation to start fcitx and pass the WAYLAND_SOCKET fd. So it won't be conflict with xdg autostart.

Anyway, if you keep only one way to start fcitx it should be fine.

**Comment by DonKatsu** [comment]
Restarted for a mesa update today and seemed to have the issue again. Just had the fcitx virtual keyboard method.
This time I had checked my hidden notification icons and saw the ibus panel. When I restarted fcitx the ibus panel icon disappeared.
Since I had set ibus panel to always hidden before, it might have been there those previous times as well.

Grepping env for fcitx showed that only `SDL_IM_MODULE=fcitx` was active. GTK_IM_MODULE, XMODIFIERS, and QT_IM_MODULE were set to IBus which was the last one I touched with im-chooser.
Trying to set `IMSETTINGS_MODULE=` to fcitx or fcitx5 wouldn't apply either. Another .conf in environment.d with `QT_LOGGING_RULES='*=false'` was still applying.

I messed with im-chooser a bit and found it was overriding those fcitx envars. No method set everything to xim and such, IBus to ibus, fcitx5 to fcitx/fcitx5. It stopped defaulting to No method when launched as well.
Without the virtual keyboard and autostart package, im-chooser no longer errored when selecting fcitx5 and it worked as expected so now I realize that's what Akira meant in that Redhat issue. And the way I recently had experienced it.

I uninstalled imsettings and after restarting my fcitx.conf worked again. Reinstalling imsettings and restarting without touching im-chooser resulted in fcitx.conf getting overridden with the values for the No method option again.
imsettings on Fedora hasn't seen an update since the upgrade to 38, and that was just a revision update.
I really don't understand what caused it to go crazy on me all of a sudden.
Uninstalled imsettings again and set virtual keyboard to fcitx5. Restarted and my fcitx.conf is set and it works properly.
[fcitx5-diagnose.txt](https://github.com/fcitx/fcitx5/files/11630977/fcitx5-diagnose.txt)

I'll close this since now it seems to be in fact an imsettings issue. Sorry again. I guess I should open a new issue on the Redhat bug tracker?
